### PR TITLE
Fixes #13946 - added clone command to config templates

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: test/unit/template_test.rb [hunk 1]
```diff
@@ -158,4 +158,17 @@
 
   end
 
+  context "CloneCommand" do
+
+    let(:cmd) { HammerCLIForeman::Template::CloneCommand.new("", ctx) }
+
+    #before :each do
+    #  cmd.stubs(:option_template_kind_id).returns(1)
+    #end
```
Comment (tstrachota) [inline test/unit/template_test.rb]: Please remove the commented code.

File: lib/hammer_cli_foreman/template.rb [hunk 2]
```diff
@@ -156,15 +156,37 @@ def print_data(message)
...
 
-    HammerCLIForeman::AssociatingCommands::OperatingSystem.extend_command(self)
-
+      success_message _('Config template cloned')
+      failure_message _('Could not clone the config template')
 
-    autoload_subcommands
+      def self.command_name(name = nil)
+        super(name) || 'clone'
+      end
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/template.rb]: Since it's less likely to inherit from this class, it can be simplified to `command_name 'clone'`

**Comment by tstrachota** [comment]
@ShimShtein lately we've started adding functional tests for commands. They enable for checking exact output and setting api expectations. They also give you much more readable message when assertions fail. Could you please add some functional tests for your code? See https://github.com/theforeman/hammer-cli-foreman/blob/master/test/functional/location_test.rb for usage details.

**Comment by ShimShtein** [comment]
Fixed the code and added functional tests. Waiting for proper API mock PR by @tstrachota

**Comment by tstrachota** [comment]
Thanks @ShimShtein, the related PRs are: https://github.com/theforeman/hammer-cli-foreman/pull/221 and https://github.com/Apipie/apipie-bindings/pull/38

We're waiting for a new release of apipie-bindings.

**Comment by tstrachota** [comment]
@ShimShtein both PRs have been merged, you can rebase and you'll get param validations in mocks.

**Comment by ShimShtein** [comment]
Rebased, and added a fix to `#api_expects_no_call`.

File: test/functional/template_test.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should print error on missing --new-name' do
+      params = ['--id=1']
+
+      expected_result = CommandExpectation.new
+      expected_result.expected_err =
+        ['Could not clone the config template:',
+         "  Missing arguments for 'config_template[name]'",
```
Comment (tstrachota) [inline test/functional/template_test.rb]: This default message isn't very helpful. Could you please change it to a message that points users to `--new-name` ? This parameter is always required so I thing the easiest way to achieve this is:
```ruby
validate_options do
  option(:option_new_name).required
end
```

**Comment by tstrachota** [comment]
@ShimShtein I'm sorry for letting you wait. I found one issue with an error message raised upon missing `--new-name`. 
Otherwise it looks good. :+1: to merge once the message is fixed.

**Comment by ShimShtein** [comment]
Added validation and fixed the test accordingly

**Comment by tstrachota** [comment]
Merged as https://github.com/theforeman/hammer-cli-foreman/commit/5aaa69fa1f9df50480002d92c9f42c5ede1b9635
Thanks @ShimShtein !

**Pull request closed without merging**